packages/algob/src/builtin-tasks/run.ts

```diff
@@ -69,11 +75,13 @@ function loadCheckpointsIntoCPData (cpData: CheckpointRepo, scriptPaths: string[
   return checkpointData;
 }
 
-// TODO: Reduce file IO:
-// Function only accepts sorted scripts -- only this way it loads the state correctly.
-// Optimization: Split the scripts into sorted array chunks and run those chunks
-// This will save some disk reads because sub-arrays will be sorted
-export async function runMultipleScriptsOneByOne (
+function batchScriptNames (unbatched: string[]): string[][] {
+  return partitionByFn(
+    (a: string, b: string) => cmpStr(a, b) === 1,
+    unbatched);
+}
```

Comment:
Please rename:

* batchScriptNames -> groupSameStrings
* partitionByFn -> groupByFn

Reply:
Re 1 I would prefer `groupIntoSorted`

Re 2. You are using a `partitionByFn` function name, which takes an argument. The function itself doesn't know anything about the order. If you would use as an argument `(a, b) => cmpStr(a, b) === 0)` then it wouldn't imply any order.

Reply:
Then the function would split when it will encounter not a smaller element but equal one:
`[1 2 2 3 4 3 2 1]`
would return
`[[1 2] [2 3 4 3 2 1]]`.
And as it runs sequentially from left to right it preserves the order which is important for me. This is not grouping and more of a splitting of a list. There is no sorting so sorted doesn't belong here. The order is not changed.

Reply:
OK, clear. Please add a comment to the function with example.

Reply:
Added this:
```
/** Partitions an unsorted string list into sorted parts:
    `[1 2 2 3 4 3 4 2 1]` returns `[[1 2 2 3 4] [3 4] [2] [1]]` */
function partitionIntoSorted (unsorted: string[]): string[][] {
  return partitionByFn(
    (a: string, b: string) => cmpStr(a, b) === 1, // split when a > b
    unsorted);
}
```